### PR TITLE
cpu/stm32f4: optimized baudrate calculation

### PR DESCRIPTION
**Pull request by haukepetersen**
The F4 does is also happy to save some ROM...

`hello-world` for `stm32f4discovery`:
```
BEFORE:
   text	   data	    bss	    dec	    hex	filename
   9212	    116	   2652	  11980	   2ecc

AFTER:
   text	   data	    bss	    dec	    hex	filename
   7900	    116	   2652	  10668	   29ac
```
Oh happy day :-)

### CODE REVIEW

File: cpu/stm32f4/periph/uart.c [hunk 1]
```diff
@@ -167,9 +166,9 @@ int uart_init_blocking(uart_t uart, uint32_t baudrate)
     }
 
     /* configure UART to mode 8N1 with given baudrate */
-    divider = clk / (16 * baudrate);
-    mantissa = (uint16_t)divider;
-    fraction = (uint8_t)((divider - mantissa) * 16);
+    clk /= baudrate;
```
Comment (PeterKietzmann) [inline cpu/stm32f4/periph/uart.c]: Won't that lead to `0 / x` somehow? Or am I mistaken?

Comment (OlegHahm) [inline cpu/stm32f4/periph/uart.c]: If no proper UART is given, yes.

**Comment by PeterKietzmann** [comment]
Looks fine. Will test next week. Unrelated Travis fails like in the other PRs.

**Comment by PeterKietzmann** [comment]
Works. ACK in general

**Comment by kaspar030** [comment]
@PeterKietzmann Show some balls! ACK because @PeterKietzmann confirmed it working.

**Comment by PeterKietzmann** [comment]
:-) but PR needs rebase

**Comment by OlegHahm** [comment]
I'm pretty sure it won't get rebased within the next weeks, but I'm fine with making an exception here.

**Comment by PeterKietzmann** [comment]
and go

**Pull request merged**